yb-voyager/src/tgtdb/target_db_interface.go

```diff
@@ -38,18 +40,89 @@ type TargetDB interface {
 	IfRequiredQuoteColumnNames(tableName string, columns []string) ([]string, error)
 	ExecuteBatch(migrationUUID uuid.UUID, batch *EventBatch) error
 	GetDebeziumValueConverterSuite() map[string]tgtdbsuite.ConverterFn
-	GetEventChannelsMetaInfo(migrationUUID uuid.UUID) (map[int]EventChannelMetaInfo, error)
-	GetTotalNumOfEventsImportedByType(migrationUUID uuid.UUID) (int64, int64, int64, error)
-	InitLiveMigrationState(migrationUUID uuid.UUID, numChans int, startClean bool, tableNames []string) error
 	MaxBatchSizeInBytes() int64
 	RestoreSequences(sequencesLastValue map[string]int64) error
-	GetImportedEventsStatsForTable(tableName string, migrationUuid uuid.UUID) (*EventCounter, error)
-	GetImportedSnapshotRowCountForTable(tableName string) (int64, error)
 	GetGeneratedAlwaysAsIdentityColumnNamesForTable(table string) ([]string, error)
 	DisableGeneratedAlwaysAsIdentityColumns(tableColumnsMap map[string][]string) error
 	EnableGeneratedAlwaysAsIdentityColumns(tableColumnsMap map[string][]string) error
+
+	// NOTE: The following four methods should not be used for arbitrary query
+	// execution on TargetDB. The should be only used from higher level
+	// abstractions like ImportDataState.
+	Query(query string) (Rows, error)
+	QueryRow(query string) Row
+	Exec(query string) (int64, error)
+	WithTx(fn func(tx Tx) error) error
+}
+
+//=============================================================
+
+type Rows interface {
+	Row
+	Next() bool
+	Close()
+}
+
+type Row interface {
+	Scan(dest ...interface{}) error
+}
+
+type sqlRowsToTgtdbRowsAdapter struct {
```

Comment:
I see the value in Tx interface, since we can define an Exec that returns both rowsAffected, err. 

here, it looks like we're just forwarding each method to the underlying sql.Rows. Is it just for future-proofing in case some other DB's pkg doesn't adhere to sql.rows?

Reply:
The `sql.Rows.Close()` method does not match what is expected by the `tgtdb.Rows` interface (which essentially is a subset of `pgx.Rows`). So the "adapter" class, defines a wrapper on the `sql.Rows.Close()` to drop the error.